homeassistant/helpers/entity_registry.py

```diff
@@ -11,22 +11,37 @@
 from disk.
 """
 import asyncio
-from collections import namedtuple, OrderedDict
+from collections import OrderedDict
 from itertools import chain
 import logging
 import os
 
+import attr
+
 from ..core import callback, split_entity_id
 from ..util import ensure_unique_string, slugify
 from ..util.yaml import load_yaml, save_yaml
 
 PATH_REGISTRY = 'entity_registry.yaml'
 SAVE_DELAY = 10
-Entry = namedtuple('EntityRegistryEntry',
-                   'entity_id,unique_id,platform,domain')
 _LOGGER = logging.getLogger(__name__)
 
 
+@attr.s(slots=True, frozen=True)
+class RegistryEntry:
+    """Entity Registry Entry."""
+
+    entity_id = attr.ib(type=str)
+    unique_id = attr.ib(type=str)
+    platform = attr.ib(type=str)
+    name = attr.ib(type=str, default=None)
+    domain = attr.ib(type=str, default=None, init=False, repr=False)
+
+    def __attrs_post_init__(self):
+        """Computed properties."""
```

Comment:
Do these behave as properties or instance attributes, ie write protected or not?

Reply:
They are write protected because the class is annotated with `frozen=True`. The class is frozen after the post_init has run. It allows me to operate on the values entered before it's frozen.

you have to read the 2 methods from `attr` as attrib and attrs btw. Makes it easier to remember